src/main/java/shop/converters/fabrics/ConvertersFabric.java

```
@@ -0,0 +1,26 @@
+package shop.converters.fabrics;
+
+import lombok.experimental.UtilityClass;
+import shop.converters.ConverterJsonToOrders;
+import shop.converters.ConverterToOrders;
+import shop.converters.ConverterTxtToOrder;
+import shop.recording.FileFormatSpecifier;
+
+@UtilityClass
+public class ConvertersFabric {
+
+    private static final ConverterToOrders DEFAULT_FORMAT = new ConverterTxtToOrder();
+
+    public static ConverterToOrders createConverter() {
+        String format = FileFormatSpecifier.determineOrdersFormat().getType();
+        switch (format) {
+            case "txt":
+                return new ConverterTxtToOrder();
+            case "json":
+                return new ConverterJsonToOrders();
+            default:
+                return DEFAULT_FORMAT;
```

You can raise the exception here with the error message like "this type is not supported"